devicetypes/smartthings/aeon-key-fob.src/aeon-key-fob.groovy

```diff
@@ -129,41 +139,50 @@ def configure() {
 
 def installed() {
 	initialize()
+	Integer buttons = (device.currentState("numberOfButtons").value).toBigInteger()
+	if(buttons > 1) {
+		createChildDevices()
+	}
 }
 
 def updated() {
 	initialize()
-	if (!childDevices) {
-		createChildDevices()
-	}
-	else if (device.label != state.oldLabel) {
-		childDevices.each {
-			def segs = it.deviceNetworkId.split("/")
-			def newLabel = "${device.displayName} button ${segs[-1]}"
-			it.setLabel(newLabel)
+	Integer buttons = (device.currentState("numberOfButtons").value).toBigInteger()
+	if(buttons > 1)
+	{
+		if (!childDevices) {
+			createChildDevices()
+		}
+		else if (device.label != state.oldLabel) {
+			childDevices.each {
+				def segs = it.deviceNetworkId.split("/")
+				def newLabel = "${device.displayName} button ${segs[-1]}"
+				it.setLabel(newLabel)
+			}
+			state.oldLabel = device.label
 		}
-		state.oldLabel = device.label
 	}
 }
 
 def initialize() {
 	// Device only goes OFFLINE when Hub is off
 	sendEvent(name: "DeviceWatch-Enroll", value: JsonOutput.toJson([protocol: "zwave", scheme:"untracked"]), displayed: false)
-
 	def zwMap = getZwaveInfo()
-	def buttons = 4 // Default for Key Fob
-
+	def buttons
 	// Only one button for Aeon Panic Button
 	if (zwMap && zwMap.mfr == "0086" && zwMap.prod == "0001" && zwMap.model == "0026") {
 		buttons = 1
 	}
-	createChildDevices(buttons)
+	else {
```

Comment:
you don't need the else block if you kept the code to what it was. the initialized value of 4